### PR TITLE
Quantize sidekiq args

### PR DESCRIPTION
**Pull request by dudo**
Allows us to strip out PII for incoming job args. This could probably replace `tag_args`, or maybe even put the hash in tag_args?? Not sure if that would cause confusion.

### CODE REVIEW

**Comment by dudo** [comment]
Couple thoughts... 

- `secret` arg _does_ seem to be obfuscated? Though that might be a red herring.
- can't seem to get the incantation correct to show the rest of the args. 
  - expected: "[\"random_id\"]"
     got: "[\"?\"]"

**Comment by delner** [comment]
Adding `quantize` and using the existing Hash quantization seems like the way to go.

Not sure about the `show: :all` obfuscating other args. I would write a unit test case for the Hash quantization module to test the options your passing have the expected result, and if it doesn't pass, it should help you debug why.

**Comment by dudo** [comment]
@delner will submit another PR regarding Hash quantization... the current code early outs with `:all`. This PR looks good to go as long as we're comfortable deprecating `tag_args`.

File: lib/datadog/tracing/contrib/sidekiq/server_tracer.rb
Comment (marcotc) [inline lib/datadog/tracing/contrib/sidekiq/server_tracer.rb:65]: @dudo any particular reason to remove Worker-level configuration?

Comment (dudo) [reply lib/datadog/tracing/contrib/sidekiq/server_tracer.rb:65]: It seems as though it was only there as a convenience method for the tests. There are no docs for it... no reference anywhere.

Lemme know if it's "real", and I'll put it back, and add something to the docs.

**Comment by delner** [comment]
Not sure about the breaking change in removing `tag_args` here. Does `quantize` provide an equivalent?

If so, we must....

 - Have `tag_args` set the equivalent `quantize` settings (so users with `tag_args` are unaffected)
 - Raise a warning when `tag_args` is accessed and suggest `quantize` should be used instead.
 - Add appropriate documentation to `GettingStarted.md`
 - Mark this to be deleted as a breaking change in 2.0.

If we can't do these, we shouldn't merge this as a breaking change.

File: lib/datadog/tracing/contrib/sidekiq/server_tracer.rb
Comment (marcotc) [reply lib/datadog/tracing/contrib/sidekiq/server_tracer.rb:65]: Keeping it for now I think is the best option.

**Comment by marcotc** [comment]
As a follow up to @delner's point:
* `tag_args: true` seems equivalent to `quantize: { show: :all }`. This means `tag_args` can be removed if we document this change. This can be done in the migration guide: https://github.com/DataDog/dd-trace-rb/blob/master/docs/UpgradeGuide.md#breaking-changes (in a new `Integrations - Sidekiq` row in the upgrade guide table)

@dudo if this is addressed, as well as restoring the per-worker configuration, I think this will be good to merge!

Thank you again for your work!

File: lib/datadog/tracing/contrib/sidekiq/server_tracer.rb
Comment (dudo) [reply lib/datadog/tracing/contrib/sidekiq/server_tracer.rb:65]: So... While putting this back, I noticed a few things that support my initial concerns. This is a private method used by this middleware. The only flags it's trying to grab are `service_name` and `tag_args`. If we really want to support dynamically setting the service, feels like we should leverage `Datadog.tracer.active_span`, no? Feels a bit more idiomatic. Based on the specs I fixed, it really feels like this was just a test helper function.

Again... no documentation about this anywhere - I'd really advocate we lose it.

**Review by marcotc — APPROVED** [review]
Thank you so much, @dudo!

**Comment by marcotc** [comment]
👋 @dudo, this change has been released in [v1.1.0](https://github.com/DataDog/dd-trace-rb/releases/tag/v1.1.0).
Thank you again! 🙇

**Comment by agrobbin** [comment]
@marcotc @dudo what is the expected behavior of these quantized arguments when using Sidekiq with a framework like Active Job, which uses a wrapper worker class to function? `job['args']` will look something like this:

```ruby
[{"job_class"=>"FooJob", "job_id"=>"18a1bfcc-8597-4715-9892-6b658e44eb07", "provider_job_id"=>nil, "queue_name"=>"default", "priority"=>nil, "arguments"=>[...], "executions"=>0, "exception_executions"=>{}, "locale"=>"en", "timezone"=>"UTC", "enqueued_at"=>"2022-07-10T03:00:35Z"}]
```

If I'm reading this code correctly, all of those "args" will be quantized and sent as job span tags, which seems unexpected.

**Comment by marcotc** [comment]
@agrobbin, you mention a very pertinent case.
If the arguments are wrapped in a known format, I believe we should handle the most common ones, like you linked in the official [Sidekiq UI](https://github.com/mperham/sidekiq/blob/ea9c86ff5052fff310e83cf51d020771795bb4b6/lib/sidekiq/api.rb#L399-L428).

I'll open an issue to track this improvement.

**Comment by marcotc** [comment]
Captured here https://github.com/DataDog/dd-trace-rb/issues/2255.

**Pull request merged**
